### PR TITLE
[CI] Build CoqIDE for macOS on Azure

### LINKED ISSUE #9427: Automatic packaging for OSX
Packages for OSX should be automatically built, as soon as a new version is tagged.

The 8.9 branch has an ad-hoc mechanism using manually-triggered builds on Gitlab-CI executed by a private trusted runner.

### PR DESCRIPTION
**Pull request by vbgl**
This PR is a work in progress: I use the CI for testing. Sorry for the noise.

This PR improves the CI pipeline for macOS on Azure.

Major changes / wish list:

 - [x] Build CoqIDE
 - [x] Create a disk image for users to download

Fixes  #9427

### CODE REVIEW

**Comment by ejgallego** [comment]
You can just remove the .gitlab-ci.yml file temporarily as to avoid cancelling manually. For reference, this is our setup in lablgtk3 upstream:

https://github.com/garrigue/lablgtk/blob/lablgtk3/.travis.yml#L30

**Comment by vbgl** [comment]
Thank you Emilio for the hint and the pointer.

File: Makefile.ide [hunk 1]
```diff
@@ -66,7 +66,7 @@ IDEFILES=$(wildcard ide/*.lang) ide/coq_style.xml ide/coq.png ide/MacOS/default_
 GTKSHARE=$(shell pkg-config --variable=prefix gtk+-3.0)/share
 GTKBIN=$(shell pkg-config --variable=prefix gtk+-3.0)/bin
 GTKLIBS=$(shell pkg-config --variable=libdir gtk+-3.0)
-PIXBUFBIN=$(shell pkg-config --variable=prefix gdk-pixbuf-3.0)/bin
+PIXBUFBIN=$(shell pkg-config --variable=prefix gdk-pixbuf-2.0)/bin
```
Comment (SkySkimmer) [inline Makefile.ide]: It was an incorrect version?

**Review by ejgallego — APPROVED** [review]
LGTM, needs testing of course.

Can you please add a link to where the artifacts for master will be stored?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -68,11 +72,36 @@ jobs:
...
+      export PKG_CONFIG_PATH=/usr/local/opt/libffi/lib/pkgconfig
+      VERSION=$(sed -n -e '/^let coq_version/ s/^[^"]*"\([^"]*\)"$/\1/p' configure.ml)
+      APP=bin/CoqIDE_${VERSION}.app
+      make PRIVATEBINARIES="$APP" "$APP"
+      make OLDROOT="$(Build.BinariesDirectory)" COQINSTALLPREFIX="$APP/Contents/Resources/" install-coq install-ide-toploop
+      mkdir -p "$DMGDIR"
+      ln -s /Applications "$DMGDIR/Applications"
+      cp -r "$APP" "$DMGDIR"
+      hdiutil create -imagekey zlib-level=9 -volname "coq-$VERSION-installer-macos" -srcfolder "$DMGDIR" -ov -format UDZO "$(Build.ArtifactStagingDirectory)/coq-$VERSION-installer-macos.dmg"
+    displayName: 'Create the dmg bundle'
```
Comment (SkySkimmer) [inline azure-pipelines.yml]: This script is a bit big, maybe put it in its own file (`azure-make-dmg.sh`?)

**Review by SkySkimmer — APPROVED** [review]

**Comment by vbgl** [comment]
For the current run of the CI, the artifact is there: https://dev.azure.com/coq/bb4af0bd-7538-469b-88fe-a614a5dc4649/_apis/build/builds/2049/artifacts?artifactName=coq-macOS-installer&api-version=5.1-preview.5&%24format=zip

Apparently there is an API so that we don’t need to guess all these random numbers…

File: Makefile.ide (hunk 1 above)
Comment (vbgl) [reply Makefile.ide]: Indeed. gdk-pixbuf-3 does not exist

**Comment by ejgallego** [comment]
> Apparently there is an API so that we don’t need to guess all these random numbers…

Usually the URLs do allow some form of rewriting for branches, so maybe we could add to the `dev/ci/README.artifacts.md` that, see what is there already.

File: azure-pipelines.yml (hunk 2 above)
Comment (vbgl) [reply azure-pipelines.yml]: Done. Thanks.

**Comment by vbgl** [comment]
I’ve tested the build result in a macOS VM running 10.12: packaging works.

However, there are a few issues that are the same as the ones on Windows: #9885.

I think that we can merge this PR and address the Gnome issues in an other one.

**Review by ppedrot — COMMENTED** [review]
Not competent to review this PR, but I can assign and merge it if needed.

**Comment by vbgl** [comment]
This would be appreciated. Thanks.

**Comment by vbgl** [comment]
どうも。

**Pull request merged**